src/prisma/generator/templates/engine/query.py.jinja

```diff
@@ -132,11 +136,20 @@ class QueryEngine(HTTPEngine):
             args.append('--enable-playground')
 
         log.debug('Starting query engine...')
+        popen_kwargs: Dict[str,Any] = {
+            "env": env,
+            "stdout": sys.stdout,
+            "stderr": sys.stderr,
+            "text": False,
+        }
+        if platform.name() != 'windows':
+            popen_kwargs["preexec_fn"] = lambda: signal.pthread_sigmask(
+                signal.SIG_UNBLOCK, [signal.SIGINT, signal.SIGTERM]
+            )
```

Comment:
So in the documentation there is a [big warning](https://docs.python.org/3/library/subprocess.html#:~:text=The%20preexec_fn%20parameter%20is%20NOT%20SAFE%20to%20use%20in%20the%20presence%20of%20threads%20in%20your%20application.%20The%20child%20process%20could%20deadlock%20before%20exec%20is%20called.) that this is not safe to use alongside threads, does this matter in our case?

I would think it doesn't right? As in our case we're just using it as an attempt to avoid deadlocks and if it deadlocks before `preexec_fn` is even called then there's nothing we can do, am I understanding this correctly?

Reply:
Correct. One could deadlock a child process when blocking signals, but I don't see how a deadlock could be produced when unblocking them. To my understanding this would be specific to multithreading environments since blocked signals may cause thread synchronization issues, but in our case we're just forking a process.